### ISSUE TITLE
sshuttle does not clear nat rules on reboot

### ISSUE THREAD

**Issue by mjs510** [sshuttle/sshuttle#629]
I've noticed that if I reboot my system without first exiting a running instance of sshuttle, the iptables NAT rules are left in place. This results in a loss of networking when the system is next booted and can be difficult to diagnose. A quick and dirty fix is simply to run ` iptables -t nat -F` (assuming you have no other nat entries you care about!).

I'm running the sshuttle process in a terminal in Gnome on Debian 10.
sshuttle is version 1.0.5.

**Comment by skuhl** [comment]
Is your OS is saving the firewall rules on shutdown and restoring them on startup? If so, perhaps it is saving the rules before the sshuttle process has exited and cleaned up the rules itself. Then, they would get restored on boot.

**Comment by mjs510** [comment]
You're quite right - I was forgetting that the nat rules that sshuttle adds wouldn't automatically be persisted. It turns out I do have iptables-persistent/netfilter-persistent installed so this is likely to just be some forgotten legacy setup on my part.

Is there anything sshuttle should do here to support this scenario (eg a plugin that impements flush)? If not, happy for this to just be closed.

**Comment by skuhl** [comment]
I'd suggest making the shutdown procedure kill sshuttle so it can cleanup its own rules before iptables-/netfilter-persistent saves the rules. One (not great, untested) hack to that would likely do the job would be a systemd service that runs early in the shutdown process and runs something like `pkill -f '.*python.*sshuttle'`
